KIPs/kip-dreaft_sdk_common_architecture.md

```diff
@@ -0,0 +1,1623 @@
+
+##### Method description
+
+| Method | Description |
+| ----------- | ----------- |
+| decode(rlpEncoded: String): ValueTransferMemo | Decodes an RLP-encoded ValueTransferMemo string, a raw transaction, and returns a ValueTransferMemo instance. It throws an exception if the decoding is failed. |
+| getRLPEncoding(): String | Returns an RLP-encoded ValueTransferMemo string. It throws an exception if the variables required for encoding are not defined. |
+| getCommonRLPEncodingForSignature(): String | Encodes and returns the values needed to sign each transaction. It throws an exception if the variables required for encoding are not defined. |
+
+#### AccountUpdate
+
+`AccountUpdate` represents a [account update transaction](https://docs.klaytn.com/klaytn/design/transactions/basic#txtypeaccountupdate). This class is implemented by extending [AbstractTransaction](#abstracttransaction).
+
+##### Variable description
+
+| Variable | Description |
+| ----------- | ----------- |
+| account: Account | An [Account](#account) instance that contains the information needed to update your account. |
+
+##### Method description
+
+| Method | Description |
+| ----------- | ----------- |
+| decode(rlpEncoded: String): AccountUpdate | Decodes an RLP-encoded AccountUpdate string, a raw transaction, and returns a AccountUpdate instance. It throws an exception if the decoding is failed. |
+| getRLPEncoding(): String | Returns an RLP-encoded AccountUpdate string. It throws an exception if the variables required for encoding are not defined. |
+| getCommonRLPEncodingForSignature(): String | Encodes and returns the values needed to sign each transaction. It throws an exception if the variables required for encoding are not defined. |
+
+#### SmartContractDeploy
+
+`SmartContractDeploy` represents a [smart contract deploy transaction](https://docs.klaytn.com/klaytn/design/transactions/basic#txtypesmartcontractdeploy). This class is implemented by extending [AbstractTransaction](#abstracttransaction).
+
+##### Variable description
+
+| Variable | Description |
+| ----------- | ----------- |
+| to: String | Address to which the smart contract is deployed. Currently, this value cannot be defined by user, so to should be defined with the default value "0x". Specifying the address will be supported in the future. |
+| value: String | The amount of KLAY in peb to be transferred to and stored in the balance of the smart contract address when the contract is initialized. If value is not defined by user, value should be defined with the default value "0x0". |
+| input: String | Data attached to the transaction. The byte code of the smart contract to be deployed and its arguments. |
+| humanReadable: String | This must be "false" since human-readable address is not supported yet. If humanReadable is not defined by user, humanReadable should be defined with the default value "false". |
```

Comment:
```suggestion
| humanReadable: String | This must be "false" since human-readable address is not supported yet. If the value is not defined by user, it is defined with the default value "false". |
```